src/domain-services/flows/flow-service.ts

```diff
@@ -322,28 +323,39 @@ export class FlowService {
     const entityCondKey = orderBy.entity as unknown;
     const entityCondKeyFlowObjectType = entityCondKey as FlowObjectType;
 
-    flowObjects = await database.flowObject.find({
-      where: {
-        objectType: entityCondKeyFlowObjectType,
-        refDirection,
-        objectID: {
-          [Op.IN]: entityIDsSorted,
-        },
-      },
-      distinct: ['flowID', 'versionID'],
-    });
+    // Order map
+    const orderMap = new Map<number, number>();
+    for (const [index, entityID] of entityIDsSorted.entries()) {
+      orderMap.set(entityID, index);
+    }
 
+    const chunks = splitIntoChunks(entityIDsSorted, 1000);
+    // Instead of doing a single query that may end up on a 'Memory Error'
+    // we will do a progressive search
+    // by chunks of 1000
```

Comment:
You should not use arbitrary size for chunks, but aim to be most efficient. The limit is much higher and [available as constant](https://github.com/UN-OCHA/hpc-api-core/blob/71f1d70db099eb59596507083c6f16c7a5cae4b4/src/util/consts.ts#L9-L14) in hpc-api-core.

It's important to note that this limit is for all the parameter replacements that Knex will do, so since you are searching with `objectType` and `refDirection` in the query as well, you should subtract 2 from this number.
```suggestion
    const chunks = splitIntoChunks(entityIDsSorted, PG_MAX_QUERY_PARAMS - 2);
```
This has become common practice we're doing lately and I'd remove the comment as it's pretty clear.